### PR TITLE
Update CHANGELOG

### PR DESCRIPTION
**Pull request by hannahramadan**
Update CHANGELOG to add feedback given in the migration guide

_Before contributing, please read our [contributing guidelines](https://github.com/newrelic/newrelic-ruby-agent/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/newrelic/.github/blob/master/CODE_OF_CONDUCT.md)._

# Overview
Describe the changes present in the pull request

Submitter Checklist:
- [ ] Include a link to the related GitHub issue, if applicable
- [ ] Include a security review link, if applicable

# Testing
The agent includes a suite of unit and functional tests which should be used to
verify your changes don't break existing functionality. These tests will run with 
GitHub Actions when a pull request is made. More details on running the tests locally can be found 
[here for our unit tests](https://github.com/newrelic/newrelic-ruby-agent/blob/main/test/README.md), 
and [here for our functional tests](https://github.com/newrelic/newrelic-ruby-agent/blob/main/test/multiverse/README.md).
For most contributions it is strongly recommended to add additional tests which
exercise your changes. 

# Reviewer Checklist
- [ ] Perform code review
- [ ] Add performance label
- [ ] Perform appropriate level of performance testing
- [ ] Confirm all checks passed
- [ ] Add version label prior to acceptance

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -52,38 +52,61 @@
...
+  Instrumentation for the following gems had been previously deprecated and has now been removed. [PR#1788](https://github.com/newrelic/newrelic-ruby-agent/pull/1788)
     - Acts As Solr
     - Authlogic
     - DataMapper
     - Rainbows
     - Sunspot
 
+  Versions of the following technologies had been previously deprecated and are no longer supported.
+
+    - CRuby: 2.2.x, 2.3.x
```
Comment (tannalynn) [inline CHANGELOG.md:81]: The ruby versions removal have their own sections in the changelog, We probably don't need to list them here too.

Comment (hannahramadan) [reply CHANGELOG.md:81]: Right lol good call

File: CHANGELOG.md [hunk 2]
```diff
@@ -52,38 +52,60 @@
...
+  The configuration option `instrumentation.thread.tracing` is now enabled by default. This configuration allows the agent to properly monitor code occurring inside threads. In Ruby agent 9.0, instrumented code within threads will be recorded and associated with the current transaction when the thread was created.
 
-- **Add Fiber Instrumentation**
+  This may be a breaking change if you are currently using custom thread instrumentation. New transactions inside of threads will no longer be started if one already exists. [PR#1767](https://github.com/newrelic/newrelic-ruby-agent/pull/1767)
+
+- **Add Fiber instrumentation**
 
   `Fiber` instances are now automatically instrumented similarly to `Thread` instances. This can be configured using `instrumentation.fiber`. [PR#1802](https://github.com/newrelic/newrelic-ruby-agent/pull/1802)
```
Comment (kaylareopelle) [inline CHANGELOG.md:63]: I don't know how it looks on the docs website, but I think GitHub now supports links within single back tick code snippets. 

All this is to say, what do you think about linking the config options to their location on our config doc?

File: CHANGELOG.md [hunk 3]
```diff
@@ -52,38 +52,60 @@
...
+    - Mongo: 1.8.x - 2.3.x
+    - Sequel: 3.37.x, 4.0.x
+    - Delayed_Job: 2.0.x - 4.0.x
+    - Sidekiq: 4.2.x
+    - Excon: below 0.55.0
+    - HttpClient: 2.2.0 - 2.8.0
+    - HttpRb: 0.9.9 - 2.2.1
+    - Typhoeus: 0.5.3 - 1.2.x
+    - Bunny: 2.0.x - 2.6.x 
+    - ActiveMerchant: 1.25.0 - 1.64.x
```
Comment (kaylareopelle) [inline CHANGELOG.md:96]: Yay! Thank you!

File: CHANGELOG.md [hunk 4]
```diff
@@ -52,38 +52,60 @@
...
+    - HttpRb: 0.9.9 - 2.2.1
+    - Typhoeus: 0.5.3 - 1.2.x
+    - Bunny: 2.0.x - 2.6.x 
+    - ActiveMerchant: 1.25.0 - 1.64.x 
+
+
+- **Updated API method `set_transaction_name`**
 
-- **Changes how the API method `set_transaction_name` works**
+  When the method `NewRelic::Agent.set_transaction_name` is called, it will now always change the name and category of the currently running transaction to what is passed into the method. This is a change from previous agent versions. [PR#1797](https://github.com/newrelic/newrelic-ruby-agent/pull/1797)
```
Comment (kaylareopelle) [inline CHANGELOG.md:101]: Should the PR link be moved to the end of the line beginning with "previously"?

**Review by kaylareopelle — COMMENTED** [review]
Found a few small things. The bug fix entry links to the PR twice.

File: CHANGELOG.md (hunk 4 above)
Comment (hannahramadan) [reply CHANGELOG.md]: [Moved](https://github.com/newrelic/newrelic-ruby-agent/pull/1827/commits/3588b245f177fbe1c38ba66ffe1ae89f36fd600d) :)

File: CHANGELOG.md (hunk 2 above)
Comment (hannahramadan) [reply CHANGELOG.md]: I'll have to look into the single back tick. But the [config option is now linked](https://github.com/newrelic/newrelic-ruby-agent/pull/1827/commits/69c5455a3b26b45b67884d14aa16cf487628e5a4). Note: the docs site isn't live yet so the link doesn't work quite yet.

**Review by kaylareopelle — APPROVED** [review]

**Pull request merged**